examples/protocol_specification_ex/output/two_party_negotiation/message.py

```diff
@@ -1,93 +1,54 @@
 """This module contains two_party_negotiation's message definition."""
 
 from enum import Enum
-from typing import Dict, FrozenSet, Optional, Set, Tuple, Union, cast
+from typing import Dict, FrozenSet, Optional, Set, Tuple, cast
 
+from aea.configurations.base import ProtocolId
 from aea.protocols.base import Message
 
 DEFAULT_BODY_SIZE = 4
 
 
-class Unit:
-    """This class represents an instance of Unit."""
-
-    def __init__(self):
-        """Initialise an instance of Unit."""
-        raise NotImplementedError
-
-    def __eq__(self, other):
-        """Compare two Unit instances."""
-        if type(other) is type(self):
-            raise NotImplementedError
-        else:
-            return False
-
-
 class DataModel:
     """This class represents an instance of DataModel."""
 
     def __init__(self):
         """Initialise an instance of DataModel."""
         raise NotImplementedError
 
-    def __eq__(self, other):
-        """Compare two DataModel instances."""
-        if type(other) is type(self):
-            raise NotImplementedError
-        else:
-            return False
-
 
-class Ali:
-    """This class represents an instance of Ali."""
+class Unit:
+    """This class represents an instance of Unit."""
 
     def __init__(self):
-        """Initialise an instance of Ali."""
+        """Initialise an instance of Unit."""
         raise NotImplementedError
 
-    def __eq__(self, other):
-        """Compare two Ali instances."""
-        if type(other) is type(self):
-            raise NotImplementedError
-        else:
-            return False
-
 
-class _IOTApp:
-    """This class represents an instance of _IOTApp."""
+class IOTApp7:
+    """This class represents an instance of IOTApp7."""
 
     def __init__(self):
-        """Initialise an instance of _IOTApp."""
+        """Initialise an instance of IOTApp7."""
         raise NotImplementedError
 
-    def __eq__(self, other):
-        """Compare two _IOTApp instances."""
-        if type(other) is type(self):
-            raise NotImplementedError
-        else:
-            return False
-
 
 class TwoPartyNegotiationMessage(Message):
     """A protocol for negotiation over a fixed set of resources involving two parties."""
 
-    protocol_id = "two_party_negotiation"
+    protocol_id = ProtocolId("fetchai", "two_party_negotiation", "0.1.0")
 
     _speech_acts = {
```

Comment:
We had said previously we'd not store this here as only the keys are required...So if anything, store the keys.

Reply:
I replied to your previous comment as well.

The reason why we added _speech_acts was to have a shorter version of the code that checks the contents of a message (under ## Check correct contents). 

Currently we spell out all performatives (CFP, PROPOSE, etc) and under each we manually state how many contents there should be (expected_nb_of_contents), and then check the message has the correct number, name and type of contents.

I used to have an implementation where you loop through the contents in the _speech_act variable (for the performative entry used in the message) and check the contents of the message match those in _speech_acts in terms of name, number and type.

We could delete the _speech_act variable, but if we ever want to simplify the code, we need to put it back in.

Reply:
Yes I am aware of that discussion. Given that the current code is what it is, we should take it out/only keep what we need. At a later stage we can think about implementing a loop. Taking it out is quicker at this stage and fully functional.